recipes/godot-cpp/all/conanfile.py

```diff
@@ -0,0 +1,141 @@
+    @property
+    def _target(self):
+        return "debug" if self.settings.get_safe("build_type") == "Debug" else "release"
+
+    @property
+    def _use_llvm(self):
+        return self.settings.get_safe("compiler") in ["clang", "apple-clang"]
+
+    @property
+    def _use_mingw(self):
+        return self._platform == "windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _libname(self):
+        return "godot-cpp.{platform}.{target}.{bits}".format(platform=self._platform, target=self._target, bits=self._bits)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _godot_headers(self):
+        return self.deps_cpp_info["godot_headers"]
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        tools.rename(glob.glob("godot-cpp-*")[0], self._source_subfolder)
+
+    def requirements(self):
+        self.requires("godot_headers/{}".format(self.version))
+
+    def configure(self):
+        minimal_cpp_standard = "14"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+
+        minimal_version = {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "10",
+            "Visual Studio": "15",
+        }
+
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler standard version support".format(self.name, compiler))
+            self.output.warn(
+                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            if compiler in ["apple-clang", "clang"]:
+                raise ConanInvalidConfiguration(
+                    "{} requires a clang version that supports the '-Og' flag".format(self.name))
+            raise ConanInvalidConfiguration(
+                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
+
+    def build(self):
+        self.run(
+            " ".join([
+                "scons",
+                "-C{}".format(self._source_subfolder),
+                "-j{}".format(tools.cpu_count()),
+                "generate_bindings=yes",
+                "bits={}".format(self._bits),
+                "headers_dir={}".format(self._headers_dir),
+                "platform={}".format(self._platform),
+                "target={}".format(self._target),
+                "use_llvm={}".format(self._use_llvm),
+                "use_mingw={}".format(self._use_mingw),
+            ])
+        )
+
+    def imports(self):
+        headers_path = os.path.join(self._source_subfolder, "godot_headers")
+        self.copy("api.json", dst=headers_path, src="res", root_package="godot_headers")
```

Comment:
Do we need to copy this? Is it not found inside the `headers_dir` directory together with all the `godot_headers`? If not, would it be better to use:

> To use an alternative `api.json` file, add `use_custom_api_file=yes custom_api_file=../api.json`. Be sure to specify the correct location where you placed your file (it can be a relative or absolute path).

Reply:
Sadly the path to the API file is not dependent on the `headers_dir` (probably a bug upstream). However, i agree it would be better to use the `custom_api_file` argument.